scipy/sparse/_csr.py

```diff
@@ -122,6 +122,70 @@ def tobsr(self, blocksize=None, copy=True):
 
     tobsr.__doc__ = _spbase.tobsr.__doc__
 
+
+    def _broadcast_to(self, shape, copy=False):        
+        if self.shape == shape:
+            return self.copy() if copy else self
+        
+        shape = check_shape(shape, allow_nd=(self._allow_nd))
+
+        both_are_1d = len(self.shape) == 1 and len(shape) == 1
+        if both_are_1d:
+            if self.shape != (1,):
+                raise ValueError(f'current shape {self.shape} cannot be'
+                             f' broadcast to new shape {shape}')
+
+            self.sum_duplicates()
+            if self.nnz == 0: # array has no non zero elements
+                return self.__class__(shape, dtype=self.dtype, copy=False)
+            
+            N = shape[0]
+            data = np.full(N, self.data[0])
+            indices = np.arange(0,N)
+            indptr = np.array([0, N])
+            return self.__class__((data, indices, indptr), shape=shape, copy=False)
+
+        if len(self.shape) == 1:
+            self = self.__class__(self.reshape(self._shape_as_2d))
+
+        old_shape = self.shape
+            
+        if len(shape) != 2:
+            raise ValueError('Target shape must be a tuple of length 2')
+
+        if len(old_shape) != len(shape):
+            raise ValueError('Original shape and target shape must have the'
+                             ' same length')
```

Comment:
I think both the approach of reshaping 1D to 2D and the approach of using `_shape_as_2d` make this check strange. The only ways it would raise is if `shape` is >2D or both `shape` is 1D and self is not.
Perhaps we should combine this check with the if statement that checks broadcasting.

Actually, can we eliminate much of this checking code by using `np.broadcast_shapes`? Is that better? Or is this better?   :}

Reply:
Ah yes, this check doesn't really make sense anymore, especially since we check that `len(shape) == 2` in the line right before this. I've removed it at https://github.com/scipy/scipy/pull/21412/commits/8eed3d09249ade78045d1dc0da82825be3b19faa.
I think using `np.broadcast_shapes` will raise the issue you pointed out a few weeks ago - `(2,1)` and `(2,2)` are broadcastable, but we can't broadcast a `(2,2)` array into a `(2,1)` array.

Reply:
You are right that (1,2) and (2,2) are broadcastable. But I thought of the idiom that would check "broadcast_to" using "broadcast_shapes":
```python
    if np.broadcast_shapes(self.shape, shape) == shape:
```
But does it have any advantages over what we're doing in terms of reducing the checks?  We still need to check a bunch...  I was also hoping it might cut down on the number of exceptions to word/reword since we could use the numpy words. But that doesn't always work well either. :)

Reply:
I moved the check above to avoid checking for the shape being `(1,)` in the 1-D case. Does this make sense? - 60050c0277eb0f1e3e83d3838209a1755925b86b
If the shapes aren't broadcastable, `np.broadcast_shapes` will throw an error during the if statement. If they are broadcastable but in the wrong order, we throw the `"cannot be broadcast"` error.